### PR TITLE
Add support rename

### PR DESCRIPTION
**Pull request by jimmyxian**
Improve rename in swarm so that it can refresh container immediately. Need this Interface
https://github.com/docker/swarm/pull/637
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by jimmyxian** [comment]
ping @vieux @ehazlett @aluzzardi

File: dockerclient.go [hunk 1]
```diff
@@ -458,3 +458,12 @@ func (client *DockerClient) Exec(config *ExecConfig) (string, error) {
 	}
 	return createExecResp.Id, nil
 }
+
+func (client *DockerClient) RenameContainer(oldName string, newName string) error {
+	uri := fmt.Sprintf("/containers/%s/rename?name=%s", oldName, newName)
+	_, err := client.doRequest("POST", uri, nil, nil)
```
Comment (vieux) [inline dockerclient.go]: You can simply `return err` here

Comment (jimmyxian) [inline dockerclient.go]: Yeah. i will modify it

**Comment by jimmyxian** [comment]
@vieux Modified it

**Comment by vieux** [comment]
LGTM

**Comment by ehazlett** [comment]
Tested -- works as expected.  Thanks for the contribution!  LGTM

**Pull request merged**